classes/Tag.php

```diff
@@ -303,13 +303,26 @@ public function setProducts($array)
      * Delete tags for product.
      *
      * @param int $idProduct Product ID
+     * @param int|null $langId provide lang id to delete tags only in specific language
      *
      * @return bool
+     *
+     * @throws PrestaShopDatabaseException
      */
-    public static function deleteTagsForProduct($idProduct)
+    public static function deleteTagsForProduct($idProduct, int $langId = null)
```

Comment:
If I use same private method to avoid duplication, i will still have the same risk, because old code will be using the same modified code anyway. So i prefer to stay with this, as its already done, is that ok?

Reply:
Yes but it clarifies the "API" we did the same thing with `TaxRulesGroup::getTaxRulesGroupsForOptions` recently I don't doubt that the code itself works so the private method is not an issue

But at least we have two distinct functions that can be used for distinct needs, instead of an optional parameter